app/controllers/users_controller.rb

```diff
@@ -0,0 +1,205 @@
+# frozen_string_literal: true
+
+# display infornation about, and edit, users
+class UsersController < ApplicationController
+  # These need to be moved into the files where they are actually used.
+  require "find"
+  require "set"
+
+  before_action :login_required
+  before_action :disable_link_prefetching
+
+  # User index, restricted to admins.
+  def index
+    return patterned_index if params[:pattern].present?
+
+    if in_admin_mode? || find_query(:User)
+      query = find_or_create_query(:User, by: params[:by])
+      show_selected_users(query, id: params[:id].to_s, always_index: true)
+    else
+      flash_error(:runtime_search_has_expired.t)
+      redirect_to("/")
+    end
+  end
+
+  alias index_user index
+  # People guess this page name frequently for whatever reason, and
+  # since there is a view with this name, it crashes each time.
+  alias list_users index
+
+  # User index, restricted to admins.
+  def by_name
+    if in_admin_mode?
+      query = create_query(:User, :all, by: :name)
+      show_selected_users(query)
+    else
+      flash_error(:permission_denied.t)
+      redirect_to("/")
+    end
+  end
+
+  # by_contribution.rhtml
+  def by_contribution
```

Comment:
See above comment about `by_name`.
Also (now paying attention to what this does) I agree with @nimmolo that -- provided that it's a separate route -- `contributors` is better than `by_contribution`.